From ee078976071c07884299512951e6739d90169970 Mon Sep 17 00:00:00 2001
From: Yue Tao <yue.tao@windriver.com>
Date: Fri, 14 Jun 2013 01:33:22 -0700
Subject: [PATCH] usb: Change the children of usb_device to child of usb_port

The children of usb_device structure was moved to to struct usb_port,
so change the codes to find the child from usb_port.

Signed-off-by: Yue Tao <yue.tao@windriver.com>
---
 drivers/usb/core/hub.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 765c057..63b0b4a 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -3941,7 +3941,7 @@ static void hub_events(void)
 			if (hub_port_warm_reset_required(hub, portstatus)) {
 				int status;
 				struct usb_device *udev =
-					hub->hdev->children[i - 1];
+					hub->ports[i - 1]->child;
 
 				dev_dbg(hub_dev, "warm reset port %d\n", i);
 				if (!udev) {
-- 
1.7.5.4

